src/pages/Store.tsx

```diff
@@ -33,25 +32,30 @@ export type LabwareAwaitingStorageInfo = {
 
 /**
  *
- * @param location - location to navigate
- * @param action - action performed
- * @param message - message to display
+ * @param nextLocation - location to navigate
+ * @param currentLocation - current location
+ * @param historyAction - action performed
  * Clear session storage for awaiting labwares if it is navigating to a page other than /location or /store
  * Session storage will be used for following operations
  * a) Go back and Go forward operation to a Location/Store page
  * b) Going to a new location page by invoking a store location link or through a search
  */
-export function awaitingStorageCheckOnExit(location: H.Location, action: H.Action, message: string) {
+export function awaitingStorageCheckOnExit(args: {
+  currentLocation: Location;
+  nextLocation: Location;
+  historyAction: HistoryAction;
+}) {
   /**PUSH is the action for  invoking/visiting a link or for pushing a new entry onto the history stack
    * POP is the action send while you navigate using the browser's forward/back buttons.
    * **/
   if (
-    (action === 'POP' && ['/locations', '/store'].some((path) => location.pathname.startsWith(path))) ||
-    (action === 'PUSH' && location.pathname.startsWith('/locations'))
+    (args.historyAction === 'POP' &&
+      ['/locations', '/store'].some((path) => args.nextLocation.pathname.startsWith(path))) ||
+    (args.historyAction === 'PUSH' && args.nextLocation.pathname.startsWith('/locations'))
   ) {
-    return true;
+    return false;
```

Comment:
It still looks different from what it was before, and I'm not sure what the change in return value means. It used to return false if (certain conditions), otherwise returns a message. Now it returns true if (certain conditions), otherwise false. What does the change in logic indicate?

Reply:
Also, what is the `currentLocation` argument for? It doesn't seem to be used.

Reply:
> Also, what is the `currentLocation` argument for? It doesn't seem to be used.

Yes, you are right. It is not currently used. But, the router library displaying the prompt dialog requires the function to be passed on to this  as a `BlockerFunction` type with the following props where currentLocation is not an optional field. So, I am afraid we cannot omit that field.

`export type BlockerFunction = (args: {
    currentLocation: Location;
    nextLocation: Location;
    historyAction: HistoryAction;
}) => boolean;`